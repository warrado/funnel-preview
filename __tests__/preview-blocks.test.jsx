import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import PreviewButton from "../app/components/preview-button";
import PreviewImage from "../app/components/preview-image";
import PreviewList, { PreviewListItem } from "../app/components/preview-list";
import PreviewPage from "../app/components/preview-page";
import PreviewText from "../app/components/preview-text";

describe("PreviewText", () => {
  it("renders", () => {
    const { container } = render(
      <PreviewText color={"blue"} align={"right"}>
        test
      </PreviewText>
    );

    const textElement = screen.queryByText("test");
    expect(textElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});

describe("PreviewButton", () => {
  it("renders unchanged", () => {
    const { container } = render(
      <PreviewButton textColor={"blue"} bgColor={"black"}>
        test button
      </PreviewButton>
    );

    const buttonElement = screen.queryByText("test button");
    expect(buttonElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});

describe("PreviewImage", () => {
  it("renders unchanged", () => {
    const { container } = render(
      <PreviewImage src={"https://via.placeholder.com/150"} alt={"test"} />
    );

    const imageElement = screen.queryByAltText("test");
    expect(imageElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});

describe("PreviewList", () => {
  it("renders all children", () => {
    const { container } = render(
      <PreviewList>
        <PreviewListItem
          title={"title1"}
          description={"description1"}
          src={"https://via.placeholder.com/150"}
        />
        <PreviewListItem
          title={"title2"}
          description={"description2"}
          src={"https://via.placeholder.com/250"}
        />
      </PreviewList>
    );

    let listItemElement = container.querySelector('[alt="title1"]');
    expect(listItemElement).toBeInTheDocument();
    listItemElement = container.querySelector('[alt="title2"]');
    expect(listItemElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("renders with no children", () => {
    const { container } = render(<PreviewList></PreviewList>);
    expect(container).toMatchSnapshot();
  });

  it("renders all children with mixed children", () => {
    const { container } = render(
      <PreviewList>
        <PreviewListItem
          title={"title1"}
          description={"description1"}
          src={"https://via.placeholder.com/150"}
        />
        <p>Test</p>
      </PreviewList>
    );

    let listItemElement = container.querySelector('[alt="title1"]');
    expect(listItemElement).toBeInTheDocument();
    listItemElement = screen.queryByText("Test");
    expect(listItemElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});

describe("PreviewPage", () => {
  it("renders unchanged", () => {
    const { container } = render(
      <PreviewPage>
        <PreviewText color={"blue"} align={"right"}>
          test
        </PreviewText>
        <PreviewButton textColor={"blue"} bgColor={"black"}>
          test button
        </PreviewButton>
        <PreviewImage src={"https://via.placeholder.com/150"} alt={"test"} />
        <PreviewList>
          <PreviewListItem
            title={"title1"}
            description={"description1"}
            src={"https://via.placeholder.com/150"}
          />
          <PreviewListItem
            title={"title2"}
            description={"description2"}
            src={"https://via.placeholder.com/250"}
          />
        </PreviewList>
      </PreviewPage>
    );
    expect(container).toMatchSnapshot();
  });
});
