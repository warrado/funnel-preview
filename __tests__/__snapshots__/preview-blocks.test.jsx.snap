// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewButton renders unchanged 1`] = `
<div>
  <button
    style="background-color: black; color: blue; border-width: 1px; border-color: rgba(0, 0, 0, 0.1); border-radius: 8px; box-shadow: 0 0 1px #00000014,0 1px 4px #0000001f; font-weight: bold; margin: 10px; padding: 12px;"
  >
    test button
  </button>
</div>
`;

exports[`PreviewImage renders unchanged 1`] = `
<div>
  <img
    alt="test"
    src="https://via.placeholder.com/150"
    style="width: 80%; border-radius: 16px;"
  />
</div>
`;

exports[`PreviewList renders all children 1`] = `
<div>
  <div
    class="grid grid-cols-3 gap-4 p-3 mt-2 mb-2"
  >
    <div
      class="flex flex-col justify-center items-center"
    >
      <img
        alt="title1"
        src="https://via.placeholder.com/150"
        style="width: 80%; border-radius: 16px;"
      />
      <p
        class="text-sm"
      >
        description1
      </p>
    </div>
    <div
      class="flex flex-col justify-center items-center"
    >
      <img
        alt="title2"
        src="https://via.placeholder.com/250"
        style="width: 80%; border-radius: 16px;"
      />
      <p
        class="text-sm"
      >
        description2
      </p>
    </div>
  </div>
</div>
`;

exports[`PreviewList renders all children with mixed children 1`] = `
<div>
  <div
    class="grid grid-cols-3 gap-4 p-3 mt-2 mb-2"
  >
    <div
      class="flex flex-col justify-center items-center"
    >
      <img
        alt="title1"
        src="https://via.placeholder.com/150"
        style="width: 80%; border-radius: 16px;"
      />
      <p
        class="text-sm"
      >
        description1
      </p>
    </div>
    <p>
      Test
    </p>
  </div>
</div>
`;

exports[`PreviewList renders with no children 1`] = `
<div>
  <div
    class="grid grid-cols-3 gap-4 p-3 mt-2 mb-2"
  />
</div>
`;

exports[`PreviewPage renders unchanged 1`] = `
<div>
  <div
    class="overflow-auto w-full h-full text-center"
  >
    <p
      style="text-align: right; font-weight: bold; padding: 12px 0px 12px 0px;"
    >
      test
    </p>
    <button
      style="background-color: black; color: blue; border-width: 1px; border-color: rgba(0, 0, 0, 0.1); border-radius: 8px; box-shadow: 0 0 1px #00000014,0 1px 4px #0000001f; font-weight: bold; margin: 10px; padding: 12px;"
    >
      test button
    </button>
    <img
      alt="test"
      src="https://via.placeholder.com/150"
      style="width: 80%; border-radius: 16px;"
    />
    <div
      class="grid grid-cols-3 gap-4 p-3 mt-2 mb-2"
    >
      <div
        class="flex flex-col justify-center items-center"
      >
        <img
          alt="title1"
          src="https://via.placeholder.com/150"
          style="width: 80%; border-radius: 16px;"
        />
        <p
          class="text-sm"
        >
          description1
        </p>
      </div>
      <div
        class="flex flex-col justify-center items-center"
      >
        <img
          alt="title2"
          src="https://via.placeholder.com/250"
          style="width: 80%; border-radius: 16px;"
        />
        <p
          class="text-sm"
        >
          description2
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`PreviewText renders 1`] = `
<div>
  <p
    style="text-align: right; font-weight: bold; padding: 12px 0px 12px 0px;"
  >
    test
  </p>
</div>
`;
